### DISCUSSION TITLE
Non-interactive (without any user prompt) credential storage on Linux

### DISCUSSION THREAD

**Discussion by bairog** [git-ecosystem/git-credential-manager#1680] [Q&A]
I have Ubuntu 22.04.4 LTS distro on my WSL2 (hosted on my Windows 11 23H2 x64 Pro).
And I need to securely and persisently store credentials in non-interactive manner (without any user prompt).
[Secret Service API](https://github.com/git-ecosystem/git-credential-manager/blob/main/docs/credstores.md#freedesktoporg-secret-service-api) definitelly require GUI, [Git credential cache](https://github.com/git-ecosystem/git-credential-manager/blob/main/docs/credstores.md#gits-built-in-credential-cachee) is not persistent, [Plaintext files](https://github.com/git-ecosystem/git-credential-manager/blob/main/docs/credstores.md#plaintext-files) are not secure at all. 
I've tried to use [GPG/pass compatible files](https://github.com/git-ecosystem/git-credential-manager/blob/main/docs/credstores.md#gpgpass-compatible-files) store. But if I set a passphrase for a generated GPG key a prompt starts to open in WSL2 terminal window (or in a GUI window if Seahorse is installed):
![image](https://github.com/user-attachments/assets/f3dac1e3-4380-454a-beae-1d1b2839d654)
So **GPG/pass compatible files** don't require any interaction only if GPG key has no passphrase? No other options available on Linux?

**Comment by becm** [comment]
In case you trust the Git instance on your Windows system, you can redirect to its [GCM credential store](https://learn.microsoft.com/en-us/windows/wsl/tutorials/wsl-git#git-credential-manager-setup).
The credentials there are protected by the Windows login, so this still only secures data-at-rest.
Be careful with this approach when using [worktrees](https://github.com/git-ecosystem/git-credential-manager/issues/1789).
